### PR TITLE
compat: fix module build on Debian with Linux 6.9

### PR DESCRIPTION
**Pull request by okt-sergeyn**
abs_srctree is unexported in Linux 6.9. It breaks compatibility scripts code on Debian. Particulary, KBUILD_SRC is obtained incorrectly. This patch replaces $abs_srctree with $srctree+KBUILD_ABS_SRCTREE=1.

---

Checked with the command line: `KVER=6.9.10-amd64 HAVE_SFC=0 make`
Please note that sfc netdriver from Onload source tree has the same issue. 

The initial error:
```
kernel_compat.sh: /usr/src/linux-headers-6.9.10-amd64 doesn't directly build x86
```

### CODE REVIEW

**Comment by okt-sergeyn** [comment]
@rhughes-xilinx please check this out. Thanks!

**Review by ivatet-amd — APPROVED** [review]
I can merge today or tomorrow if @rhughes-xilinx doesn't object.

File: src/driver/linux_resource/kernel_compat_funcs.sh [hunk 1]
```diff
@@ -547,14 +547,25 @@ vmsg "KPATH      := $KPATH"
 #  CONFIG_PTP_1588_CLOCK: PTP clock support
 
 [ -n "${ARCH:-}" ] && export ARCH
-eval $(read_make_variables KBUILD_SRC ARCH SRCARCH CONFIG_X86_32 CONFIG_X86_64 CONFIG_PTP_1588_CLOCK abs_srctree)
+eval $(read_make_variables KBUILD_SRC ARCH SRCARCH CONFIG_X86_32 CONFIG_X86_64 CONFIG_PTP_1588_CLOCK)
+
+# linux-6.9: $abs_srctree is unexported. It breaks the code below on Debian,
+# due to its kernel headers path split for "arch" and "common" ones.
+# Use $srctree with KBUILD_ABS_SRCTREE=1 instead.
+eval $(EXTRA_MAKEFLAGS="$EXTRA_MAKEFLAGS KBUILD_ABS_SRCTREE=1" read_make_variables srctree)
```
Comment (rhughes-xilinx) [inline src/driver/linux_resource/kernel_compat_funcs.sh:555]: Did you try merging this with the previous `read_make_variables` (i.e. both would run with `KBUILD_ABS_SRCTREE=1`)? Just trying to save a few milliseconds of build time

**Review by rhughes-xilinx — APPROVED** [review]

File: src/driver/linux_resource/kernel_compat_funcs.sh (hunk 1 above)
Comment (okt-sergeyn) [reply src/driver/linux_resource/kernel_compat_funcs.sh]: Fixed

**Pull request merged**